bindings/csharp/Grandine.NethermindPlugin/GrandinePlugin.cs

```diff
@@ -0,0 +1,158 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Autofac;
+using Nethermind.Api;
+using Nethermind.Api.Extensions;
+using Nethermind.Core;
+using Nethermind.Core.Authentication;
+using Nethermind.Core.Timers;
+using Nethermind.JsonRpc;
+using Nethermind.Logging;
+using Nethermind.Merge.Plugin;
+using Grandine.NethermindPlugin;
+using Nethermind.Api.Steps;
+using Nethermind.Consensus;
+using Nethermind.Consensus.Producers;
+using Nethermind.Consensus.Transactions;
+using Nethermind.Merge.Plugin.BlockProduction;
+using Nethermind.Merge.Plugin.GC;
+using Nethermind.Merge.Plugin.Handlers;
+using Nethermind.JsonRpc.Modules;
+using Nethermind.Config;
+using Nethermind.Blockchain.Synchronization;
+using Nethermind.Merge.Plugin.InvalidChainTracker;
+using Nethermind.Blockchain;
+using Nethermind.Blockchain.Receipts;
+using Nethermind.Consensus.Rewards;
+using Nethermind.Consensus.Validators;
+using Nethermind.Facade.Eth.RpcTransaction;
+using Nethermind.Merge.Plugin.Synchronization;
+using Nethermind.HealthChecks;
+using Nethermind.Init.Steps;
+using Nethermind.Specs.ChainSpecStyle;
+using Nethermind.Blockchain.Find;
+using Nethermind.JsonRpc.Modules.Eth;
+
+namespace Grandine.NethermindPlugin;
+
+public class GrandinePlugin(IGrandineConfig grandineConfig) : INethermindPlugin
+{
+    public string Name => "Grandine plugin";
+    public string Description => "Nethermind plugin to enable embedded grandine CL client";
+    public string Author => "Grandine team";
+
+    private ILogger _logger;
+
+    #pragma warning disable CS8618 
+    private INethermindApi _api;
+
+    public bool Enabled => grandineConfig.Enabled ?? throw new ArgumentNullException(nameof(grandineConfig.Enabled));
+
+    private List<string> _arguments = new List<string>();
+
+    public Task Init(INethermindApi nethermindApi)
+    {
+        _api = nethermindApi;
+        _logger = nethermindApi.LogManager.GetClassLogger();
+        _logger.Warn("Initializing grandine plugin...");
+
+        Type configType = grandineConfig.GetType();
+
+        var rpcConfig = nethermindApi.Config<IJsonRpcConfig>();
+
+        var configInterface = configType.GetInterface("IGrandineConfig");
+
+        if (configInterface == null)
+        {
+            throw new Exception("Unexpected exception occurred - IGrandineConfig interface not found");
+        }
```

Comment:
This is weird, if you loaded your plugin, you do have the types loaded correctly - so seems not needed.